<a href="https://colab.research.google.com/github/bigirimanainnocent12/Test-de-KHI-DEUX/blob/main/Code_Test_de_khi_deux.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ***Test de khi deux d'indépendances***

L'objectif de cette analyse statistique est de déterminer si la couleur des cheveux est indépendante du sexe de l'indivudu dans un échantillon composé de garçons et de filles d'un district écossais. Pour ce faire, nous allons utiliser un test d'indépendance approprié afin de vérifier s'il existe une relation significative entre la couleur des cheveux et le sexe des individus observés.

En analysant les données présentées dans le tableau ci-dessous, nous pourrons évaluer si la répartition des couleurs de cheveux diffère de manière significative entre les garçons et les filles de cet échantillon.

In [1]:
import pandas as pd
import numpy as np
import scipy.stats as stats

obs=np.array([[592,119,849,504,36],[544,97,677,451,14]])
# Transpose the obs array to have shape (5, 2) to match the index and columns
df=pd.DataFrame(obs, index=list(["HOMME", "FEMME"]), columns=list(["BLOND", "ROUX", "CHATAIN","BRUN","NOIR"]))
df

BLOND  ROUX  CHATAIN  BRUN  NOIR
HOMME    592   119      849   504    36
FEMME    544    97      677   451    14

# ***Variables***
1. Couleur des cheveux (variable qualitative nominale à 5 modalités) :

* Blond
* Roux
* Châtain
* Brun
* Noir
2. Sexe (variable qualitative nominale à 2 modalités) :

* Masculin
* Féminin

# **Hypothèses du test**

Afin de déterminer si le sexe des individus d’un district écossais a une influence significative sur la couleur de leurs cheveux, nous formulerons les hypothèses suivantes:

$H_0$ : Le sexe de l’individu d’un district écossais n’a pas d’influence significative sur sa couleur des cheveux.

$H_1$: Le sexe de l’individu d’un district écossais a une influence significative sur sa couleur des cheveux.


# **Choix du test**


Pour déterminer si la couleur des cheveux est indépendante du sexe, nous pouvons utiliser un test statistique appelé le test du Chi-2 (chi-carré) d'indépendance qui utilise une approximation d’un ratio de log-vraisemblance. Ce test permet de vérifier s'il y a une relation significative entre deux variables catégorielles, en l'occurrence ici la couleur des cheveux et le sexe. Mais nous allons utiliser le G test qui est considéré comme plus puissant que le test du χ² parce que ce test calcule le vrai rapport de log-vraisemblance ce qui nous permet d’obtenir des résultats plus fiables.



# **Choix du seuil**

Le seuil en statistique est une valeur critique qui est utilisée pour prendre une décision concernant une hypothèse statistique. Il peut être défini comme la limite au-delà de laquelle une observation est considérée comme significative ou non significative. Dans le cadre de cet exercice, nous avons établi le seuil à 5% pour évaluer l’influence du sexe sur la couleur des cheveux des individus d’un district écossais.

# **Conditions d'application du G test**

1. Notre échantillon est composé des individus qui ont été choisis de manière aléatoirement.

2. Chaque individu appartient dans une seule classe.

3. Règle de Cochran : au moins 80% des effectifs théoriques sont au moins égaux à 5.


In [2]:
from scipy import stats
G,p,dof,expected=stats.chi2_contingency(df,correction=False,lambda_="log-likelihood")
expected1=pd.DataFrame(expected, index=list(["HOMME", "FEMME"]), columns=list(["BLOND", "ROUX", "CHATAIN","BRUN","NOIR"]))
expected1

BLOND        ROUX     CHATAIN        BRUN       NOIR
HOMME  614.370332  116.816894  825.289724  516.482101  27.040948
FEMME  521.629668   99.183106  700.710276  438.517899  22.959052

Nous constatons que toutes les fréquences théoriques sont supérieures à 5, ce qui permet de vérifier la règle de Cochran. Par conséquent, nous utiliserons le G-test pour tester l’indépendance entre les deux variables.

# **Résulats du test (G-Test)**

In [5]:
dic={"Statistique de test":G,"DF":dof,'p-value F':p}
test=pd.DataFrame(dic,index=["G-Test"])
test

Statistique de test  DF  p-value F
G-Test            10.755513   4   0.029454

La statistique de test G= 10.7555. Sous $H_o$, elle suit une loi de Khi-deux à 4 degrés de liberté. Ainsi la p-value =0,0295 est statistiquement significative donc inférieure à 0,05. Nous rejetons $H_O$ au profit de $H_1$ et nous concluons que Le sexe de l’individu d’un district écossais a une influence significative sur sa couleur des cheveux.

# **Conclusion**

Les deux variables (couleur des cheveux et sexe) sont dépendantes. Nous pouvons donc affirmer que le sexe des individus d’un district écossais a un effet significatif sur la variation de la couleur de leurs cheveux.